source/nodejs/adaptivecards-site/service-worker.js

```diff
@@ -5,6 +5,14 @@ workbox.setConfig({
   modulePathPrefix: "node_modules/workbox/"
 });
 
+self.addEventListener("install", (e) => {
+     self.skipWaiting();
+})
+
+self.addEventListener("activate", (e) => {
+    e.waitUntil(clients.claim());
```

Comment:
Look like it's a nice to have but not a must have.

I'm wondering what's penalty here. With the new change, and if claim is not called, will the service worker check the network more often?

```
When a service worker is initially registered, pages won't use it until they next load. The claim() method causes those pages to be controlled immediately. Be aware that this results in your service worker controlling pages that loaded regularly over the network, or possibly via a different service worker.
```

Reply:
@licanhua From my understanding, this piece of code ensures that you don't need to refresh a page in order for the service worker to take over. I'm not sure what the behavior is if you call `skipWaiting` without `claim`.

I spoke to Aaron, and he said the major concern with this is if your activate or install events include major data structure changes. I don't believe we have any changes that would cause an issue, so I went ahead and added this piece.

That being said, the caching strategy updates greatly improved the service worker, and this might not be necessary.

@aarongustafson Do you have any thoughts here?

Reply:
As I understand it, without `claim()`, you run the risk of a page potentially running without an active SW or with an out of date one. So if someone had two tabs open to the Designer, the first one might trigger a SW update and activation, taking over Tab 1, but Tab 2 would still be using the old SW until refreshed. Claiming all clients immediately on activation ensures any currently runnign tabs & windows all get the (new) SW ASAP.

Reply:
@licanhua

Reply:
Likely perf is not a concern here. Look's good to me